msteams-platform/build-your-first-app/build-and-run.md

```diff
@@ -1,80 +1,199 @@
+1. Select **Join Now** and follow the onscreen instructions.
+1. In the Welcome screen, select **Set up E5 subscription**.
+1. Set up your administrator account. 
+   After you finish, the following screen appears:
+
+   :::image type="content" source="../assets/images/build-your-first-app/dev-program-subscription.png" alt-text="Example of what you see after signing up for the Microsoft 365 developer program.":::
+
+1. Sign in to Teams using your administrator account.
+1. In the Teams client, select **Apps**.
+1. Verify that **Upload a custom app** appears.
+
+   :::image type="content" source="../assets/images/build-your-first-app/upload-custom-app-closeup.png" alt-text="Illustration showing where in Teams you can upload a custom app.":::
+
+# [Have a tenant and admin access](#tab/have-a-tenant)
+
+If you already have a tenant with admin access to the Teams client, verify if you can sideload your apps.
+
+**Verify that you can sideload your apps** 
+
+1. In the Teams Client, select **Apps**. 
+1. Verify that **Upload a custom app** appears. 
+
+   :::image type="content" source="../assets/images/build-your-first-app/upload-custom-app-closeup.png" alt-text="Illustration showing where in Teams you can upload a custom app.":::
+    
+---
+### Install your development tools
+
+To help you get started quickly, Microsoft provides development tools such as Microsoft Teams Toolkit for Visual Studio Code. However, you can build Teams apps with your preffered tools as well.
+
+> [!NOTE]
+> Teams displays app content only through HTTPS connections. To debug certain types of apps locally, such as a bot, you'll learn how to use ngrok to set up a secure tunnel between Teams and your app.
+> 
+> Production Teams apps are hosted in the cloud.
+
+**To install Microsoft Teams Toolkit**
+
+1. Install [Node.js](https://nodejs.org/en/).
+1. Install [ngrok](https://ngrok.com/download) if you plan to build a bot or messaging extension and [create a tunnel using ngrok](../tutorials/get-started-dotnet-app-studio.md/#Tunnel using ngrok).
+1. Install the latest version of [Visual Studio Code](https://code.visualstudio.com/download). 
+   > [!NOTE]
+   > The toolkit does not support the earlier versions of Visual Studio Code.
+
+1. In the left activity bar, select **Extensions** :::image type="icon" source="../assets/icons/vs-code-extensions.png":::.
+1. In **Microsoft Teams Toolkit**, select **Install**.
+
+   :::image type="content" source="../assets/images/build-your-first-app/vsc-install-toolkit.png" alt-text="Illustration showing where in Visual Studio Code you can install the Microsoft Teams Toolkit extension.":::
 
 ## 1. Create your app project
 
-Use the Microsoft Teams Toolkit in Visual Studio Code to set up your first app project. Create your app project using the following steps:
+1. Open Visual Studio Code.
+1. Select **Microsoft Teams Toolkit** :::image type="icon" source="../assets/icons/vsc-toolkit.png"::: > **Create a new Teams app**.
```

Comment:
```suggestion
1. Select **Microsoft Teams Toolkit** :::image type="icon" source="../assets/icons/vsc-toolkit.png"::: > **Create a new Teams app**.
```
Please replace the icon with this version: 
![vsc-toolkit png](https://user-images.githubusercontent.com/66389316/115794929-c79e0f80-a383-11eb-8446-61c2410574b6.png)